2022/10/ssh-askpass-on-Mac-OS-for-agent-confirmation.md

```diff
@@ -0,0 +1,106 @@
+---
+author: "Bharathi Ponnusamy"
+date: 2022-10-26
+title: "Ssh-askpass on Mac OS for the agent confirmation"
+github_issue_number: 
+tags:
+- ssh 
+- ssh-askpass
+- mac
+- security
+---
+
+![On a summer evening](/blog/2022/10/ssh-askpass-on-Mac-OS-for-agent-confirmation/banner.jpg)
+[Photo](https://flic.kr/p/2nUPsJQ) by [Kristoffer Trolle](https://www.flickr.com/people/kristoffer-trolle/), [CC BY 2.0](https://creativecommons.org/licenses/by/2.0/)
+
+ssh-askpass for OS X/macOS. Works in (at least) 10.7+ (including Monterey), this also works on the latest M1 pro chip.
+
+Used to accept (or deny) the use of the private key(s) added to the SSH authentication agent with `ssh-add -c`.
+
+At End Point Dev we mostly use the SSH keys for authentication when connecting to remote servers and Git services. The majority of the time, the servers we are trying to visit are barred from direct access and require a middle "jump server" instead.
+
+We use SSH key forwarding, which enables us to use the private key kept on our local machine to authenticate with each server in the chain without entering a password.
+
+Using ssh-askpass as the agent confirmation is one of the recommended way when using SSH key forwarding. When utilizing the ssh-askpass, a prompt will show on your local computer to confirm the usage of the key each time a request is made to use the private key loaded in the SSH agent. Doing this makes it harder for an attacker to utilize the private key without permission.
+
+If case if you don’t want to using agent confirmation, then it’s best to use ProxyJump instead of agent forwarding.
+
+![agent confirmation dialog](/blog/2022/10/ssh-askpass-on-Mac-OS-for-agent-confirmation/ssh-askpass.png)
+
+## Installation
+
+### [Homebrew](https://brew.sh/)
+You can install ssh-askpass using Homebrew, this method works on the Apple M1 Pro chip as well.
+
+* Install:
+
+```
+    $ brew tap theseal/ssh-askpass
+    $ brew install ssh-askpass
+
+```
+* Follow caveats
+
+### [MacPorts](https://www.macports.org)
+* Install:
+
+```
+    $ sudo port install ssh-askpass
+```
+
+### Without Homebrew/MacPorts
+
+* Run:
+
+```
+$ cp ssh-askpass /usr/local/bin/
+$ cp ssh-askpass.plist ~/Library/LaunchAgents/
+$ launchctl load -w ~/Library/LaunchAgents/ssh-askpass.plist
+```
+
+## Install X11R6 
+ System Integrity Protection is a good security feature and doesn't need to be disabled in macOS 10.13+ to keep ssh_askpass working
+
+* (Re)install xquartz:
+This will officially recreate the /usr/X11R6 directory without disturbing SIP.
+
+```
+$ brew reinstall xquartz
+```
+
+* Create Symbolic Links:
+
+```
+$ cd /usr/X11R6/bin
+
+$ sudo ln -s /usr/local/bin/ssh-askpass ssh-askpass
+or 
+$ sudo ln -s `which ssh-askpass` ssh-askpass
+```
```

Comment:
@bharathi3p Why are you installing xquartz/X11? I've got the same ssh-askpass installed on macOS and don't have X there at all. Are you sure that is needed, or is it possible it was only needed for some other thing?

Reply:
@jonjensen On the older version it didn't work well as expected, and we have installed the quartz instead of disabling the System Integrity Protection.

Looks like the X11/Xquartz doesn't need to be on the newer version of mac OS.

I have tested the same on my mac OS - `Monterey 12.6` and Kannan's `M1 chip mac OS with Ventura`

Reply:
Removed the Xquartz steps and added note to start the ssh-askpass service.

Reply:
@bharathi3p Excellent. Thanks.

@sethjensen1 Can you take it from here?